lib/Semantic_check.ml

```diff
@@ -19,6 +19,17 @@ open Pretty_printing
    on (1) a global symbol table vm, and (2) structure of type context_flags_record
    to communicate information down the AST. *)
 
+(** Origin blocks, to keep track of where variables are declared *)
+type originblock =
```

Comment:
I see `originblock` moved to here, but the thing with doc in the other file commented above didn't seem to have an origin block in it.

Reply:
Sorry. That comment was outdated.